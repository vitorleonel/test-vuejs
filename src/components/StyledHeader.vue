<script lang="ts">
import { defineComponent } from 'vue'

export default defineComponent({})
</script>

<template>
  <header class="header">
    <div class="container">
      <h1>BestBuy</h1>

      <nav>
        <router-link to="/">Home</router-link>
        <router-link to="/shop">Shop</router-link>
        <router-link to="/about">About us</router-link>
      </nav>
    </div>
  </header>
</template>

<style lang="scss" scoped>
.header {
  height: 64px;
  background-color: var(--primary-color);
  display: flex;
  align-items: center;
  flex-shrink: 0;
  box-shadow: 0 2px 6px rgba(0, 0, 0, 0.15);
  position: sticky;
  top: 0;
  z-index: 10;

  .container {
    display: flex;
    align-items: center;
    justify-content: space-between;
    gap: 32px;
  }

  h1 {
    font-size: 24px;
    text-transform: lowercase;
  }

  nav {
    display: flex;
    gap: 32px;

    a {
      text-transform: lowercase;
      color: var(--white-color);
    }
  }
}
</style>
