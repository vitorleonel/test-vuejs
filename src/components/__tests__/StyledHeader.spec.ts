import { mount, type VueWrapper } from '@vue/test-utils'
import { describe, expect, it } from 'vitest'

import StyledHeader from '@/components/StyledHeader.vue'

describe('StyledHeader', () => {
  it('should render properly', () => {
    const wrapper: VueWrapper = mount(StyledHeader, { global: { stubs: ['RouterLink'] } })

    expect(wrapper.element).toMatchSnapshot()
  })
})
