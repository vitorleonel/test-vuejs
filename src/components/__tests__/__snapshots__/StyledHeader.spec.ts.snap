// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`StyledHeader > should render properly 1`] = `
<header
  class="header"
  data-v-51353816=""
>
  <div
    class="container"
    data-v-51353816=""
  >
    <h1
      data-v-51353816=""
    >
      BestBuy
    </h1>
    <nav
      data-v-51353816=""
    >
      <router-link
        data-v-51353816=""
        to="/"
      >
        Home
      </router-link>
      <router-link
        data-v-51353816=""
        to="/shop"
      >
        Shop
      </router-link>
      <router-link
        data-v-51353816=""
        to="/about"
      >
        About us
      </router-link>
    </nav>
  </div>
</header>
`;
